mobgap/Reorientation/CorrectSensorOrientationDynamic.py

```diff
@@ -0,0 +1,175 @@
+    sig3 = (newacc[:, 2] - np.mean(newacc[:, 2])) / np.std(newacc[:, 2])
+    sig4 = (av[:, 1] - np.mean(av[:, 1])) / np.std(av[:, 1])
+
+    # Assigning av_pca and gyr_pca
+    av_pca = av_magpca.copy()
+    gyr_pca = gyr_magpca.copy()
+
+    # Calculating correlations
+    cor1 = np.dot(sig1.T, sig2)
+    cor2 = np.dot(sig1.T, sig3)
+
+    if abs(cor1) > abs(cor2):
+        if cor1 > 0:
+            av_pca[:, 2] = newacc[:, 1]  # AP
+            gyr_pca[:, 2] = newgyr[:, 1]
+        else:
+            av_pca[:, 2] = -newacc[:, 1]  # AP
+            gyr_pca[:, 2] = newgyr[:, 1]
+
+        if np.dot(sig3.T, sig4) > 0:
+            av_pca[:, 1] = newacc[:, 2]  # ML
+            gyr_pca[:, 1] = newgyr[:, 2]
+        else:
+            av_pca[:, 1] = -newacc[:, 2]  # ML
+            gyr_pca[:, 1] = newgyr[:, 2]
+    else:
+        if cor2 > 0:
+            av_pca[:, 2] = newacc[:, 2]  # AP
+            gyr_pca[:, 2] = newgyr[:, 2]
+        else:
+            av_pca[:, 2] = -newacc[:, 2]  # AP
+            gyr_pca[:, 2] = newgyr[:, 2]
+
+        if np.dot(sig2.T, sig4) > 0:
+            av_pca[:, 1] = newacc[:, 1]  # ML
+            gyr_pca[:, 1] = newgyr[:, 1]
+        else:
+            av_pca[:, 1] = -newacc[:, 1]  # ML
+            gyr_pca[:, 1] = newgyr[:, 1]
+
+    # Refinement to provide the IMU axes as standard data matrix
+    av_pca_final = av_pca.copy()
+    gyr_pca_final = gyr_pca.copy()
+
+    #  Savitzky-Golay filter
+    av_pca_filt = np.zeros_like(av_pca)
+
+    savgol_win_size_samples = 11
+    savgol = SavgolFilter(
+        window_length_s=savgol_win_size_samples / sampling_rate_hz,
+        polyorder_rel=0 / savgol_win_size_samples,
+    )
+
+    filter_chain = [("savgol", savgol)]
+
+    av_pca_filt[:, 2] = chain_transformers(av_pca[:, 2], filter_chain, sampling_rate_hz=sampling_rate_hz)
+    av_pca_filt[:, 1] = chain_transformers(av_pca[:, 1], filter_chain, sampling_rate_hz=sampling_rate_hz)
+    av_pca_filt[:, 0] = chain_transformers(av_pca[:, 0], filter_chain, sampling_rate_hz=sampling_rate_hz)
+
+    # Standardization
+    sig5 = (av_pca_filt[:, 2] - np.mean(av_pca_filt[:, 2])) / np.std(av_pca_filt[:, 2])
+    sig6 = (av_pca_filt[:, 1] - np.mean(av_pca_filt[:, 1])) / np.std(av_pca_filt[:, 1])
+    sig7 = (av_pca_filt[:, 0] - np.mean(av_pca_filt[:, 0])) / np.std(av_pca_filt[:, 0])
+
+    # Cross-correlation
+    r1 = correlate(sig7, sig5)
+    r2 = correlate(sig7, sig6)
+
+    if np.max(r1) > np.max(r2):
+        av_pca_final[:, 2] = av_pca[:, 2]
+        av_pca_final[:, 1] = av_pca[:, 1]
+
+        gyr_pca_final[:, 2] = gyr_pca[:, 2]
+        gyr_pca_final[:, 1] = gyr_pca[:, 1]
+
+    else:
+        av_pca_final[:, 2] = av_pca[:, 1]
+        av_pca_final[:, 1] = av_pca[:, 2]
+
+        gyr_pca_final[:, 2] = gyr_pca[:, 1]
+        gyr_pca_final[:, 1] = gyr_pca[:, 2]
+
+    IMU_corrected = np.concatenate((av_pca_final, gyr_pca_final), axis=1)
+    IMU_corrected = pd.DataFrame(IMU_corrected)
```

Comment:
This dataframe does not have the original names and missing the index. I would suggest to try to keep the dataframe for most of the operation, instead of switching to numpy dataframes

Reply:
I updated the function to use dataframes so named access is used now everywhere